app/views/layouts/application.html.erb

```diff
@@ -19,6 +19,7 @@
         </div>
       </div>
       <% if current_user %>
+        <%= link_to current_user.name, user_path(current_user) %>
         <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
       <% else %>
         <%= link_to 'Sign in', user_session_path %>
```

Comment:
There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.